cpp/src/arrow/csv/parser.cc

```diff
@@ -359,6 +359,18 @@ Status BlockParser::ParseLine(ValuesWriter* values_writer, ParsedWriter* parsed_
   return Status::OK();
 
 EmptyLine:
+  if (!options_.ignore_empty_lines) {
+    if (num_cols_ == -1) {
+      // Consider as single value
```

Comment:
I'm not sure what the sane behavior is, my intuition tells to skip them. And it seems pandas does the same:

```
In [30]: pd.read_csv(io.BytesIO(b"""\n\nab,cd\n12,34\n\r\n56,78\n"""))
Out[30]: 
   ab  cd
0  12  34
1  56  78
```

Reply:
@fsaintjacques See the example I posted on JIRA. It skips them by default, but you can tell it to turn them to nulls.

Reply:
I'm pointing about inferring the number of columns from the empty lines. My intuition is, if the file start with empty lines, infer the number/name of columns on the first non-empty line or error, regardless of the option. It seems that pandas refuse the file in this case
```
----> 1 pd.read_csv(io.BytesIO(b"""\n\n\nab,cd\n12,34\n\r\n56,78\n"""), skip_blank_lines=False)
...
pandas/_libs/parsers.pyx in pandas._libs.parsers.TextReader.__cinit__()

EmptyDataError: No columns to parse from file
```

Reply:
Ah, so in pandas, the empty lines are only skipped after finding the header (so empty lines at the beginning need to be skipped explicitly, otherwise the first row (which is empty) is being used ad header, resulting in 0 columns.

Reply:
I'll merge for now, if it gets requested, we'll open an issue.